Introduce Lua 5.4 application to catalog-core

Dana-Maria04 opened the pull request:
### What’s included
- **Config.uk** & **Makefile.uk**  
  Unikraft-specific configuration and build scripts for the Lua unikernel
- **Makefile** & **setup.sh**  
  Top-level wrappers to bootstrap the build (`make all`, `make distclean`, etc.)
- **helloworld.lua** in **rootfs/**  
  A minimal Lua script that prints “Hello from Lua!”
- **README.md**  
  How to configure, build and run (QEMU/Xen) the Lua unikernel
- **Defconfigs**  
  - `fc.x86_64.json`, `fc.arm64.json` for functional tests  
  - `xen.x86_64.cfg`, `xen.arm64.cfg` for Xen targets

razvand commented on lua/Config.uk:
```diff
@@ -0,0 +1,15 @@
+config APPLUA
```
Improve this with comments. See example for `Nginx`.

razvand commented on lua/Makefile:
```diff
@@ -0,0 +1,56 @@
+# UK_ROOT ?= $(PWD)/workdir/unikraft
```
Remove commented rules.

razvand commented on lua/README.md:
```diff
@@ -0,0 +1,179 @@
+  -kernel workdir/build/lua_qemu-x86_64 \
+  -append "lua_qemu-x86_64 vfs.fstab=[\"initrd0:/:extract::ramfs=1:\"] -- /helloworld.lua" \
+  -initrd ./initrd.cpio
+
+
```
```suggestion
```

Remove unnecessary empty lines.

razvand commented on lua/README.md:
````diff
@@ -0,0 +1,179 @@
+
+```
+
+This will configure, build and run the Lua app, printing “Hello, World!” from helloworld.lua.
````
```suggestion
This will configure, build and run the Helloworld Lua app, printing “Hello, World!”.
```

razvand commented on lua/README.md:
````diff
@@ -0,0 +1,179 @@
+./setup.sh
+```
+
+It will clone/link lib-lua and any other needed libs under repos/libs/.
````
```suggestion
It will clone/link `lib-lua` and any other needed libs under `repos/libs/.`
```

razvand commented on lua/README.md:
```diff
@@ -0,0 +1,179 @@
+
+## Set Up
+
+Set up the required repositories. You have two options:
```
```suggestion
Set up the required repositories.
You have two options:
```

One sentence per line.

razvand reviewed (changes requested):
Use a single commit. Squash the two current commits into a single one.

Dana-Maria04 commented:
All feedback applied !

razvand commented on lua/.gitignore:
```diff
@@ -0,0 +1,19 @@
+# Build output directories
+workdir/
+build/
+
+# Kconfig-generated files
+*.cmd
+*.o
+*.d
+*.confdata
+*.tab.c
+*.tab.h
+
+# Unikraft specific
+fc.*.json
+
+# Editor / OS junk
+*.swp
+*~
+.DS_Store
```
Use the `.gitignore` file from [Nginx](https://github.com/unikraft/catalog-core/blob/main/nginx/.gitignore).

razvand commented on lua/Config.uk:
```diff
@@ -0,0 +1,29 @@
+	select LIBPOSIX_ENVIRON
+	# Use library-parameter parsing for environment variables.
+	select LIBPOSIX_ENVIRON_LIBPARAM
+	
```
Remove extra ending newline.

razvand commented on lua/README.md:
```diff
@@ -0,0 +1,172 @@
+./setup.sh
+make distclean
+wget -O /tmp/defconfig \
+     https://raw.githubusercontent.com/unikraft/app-lua/stable/defconfigs/qemu-x86_64-initrd
```
Use the same link as below: https://raw.githubusercontent.com/unikraft/catalog-core/refs/heads/scripts/lua/scripts/defconfig/qemu.x86_64

razvand commented on lua/fc.arm64.json:
```diff
@@ -0,0 +1,12 @@
+{
```
Indent JSON.

razvand commented on lua/fc.x86_64.json:
```diff
@@ -0,0 +1,12 @@
+{
```
Indent JSON.

razvand commented on lua/setup.sh:
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+check_exists_and_create_symlink() {
```
Do not indent using Tabs, indent using spaces.

Place curly bracket on a newline.

razvand commented on lua/README.md:
```diff
@@ -0,0 +1,183 @@
+
+This will configure, build and run the Helloworld Lua app, printing “Hello, World!”.
+
+To do the same for AArch64, run:
```
```suggestion
To do the same for AArch64, run:

```

Add newline before and after code snippets.

razvand commented on lua/README.md:
````diff
@@ -0,0 +1,183 @@
+qemu-system-aarch64 -nographic -machine virt -m 128 -cpu max \
+  -kernel workdir/build/lua_fc-arm64
+```
+Similar to the x86_64 build, this will print “Hello, World!” on ARM64.
````
```suggestion

Similar to the x86_64 build, this will print “Hello, World!” on ARM64.
```

Add newline before and after code snippets.

razvand commented on lua/.gitignore:
```diff
@@ -0,0 +1,3 @@
+/rootfs/
+/workdir/
+initrd.cpio
```
```suggestion
```

Not required. It's ignored by the [top-level `.gitignore` file](https://github.com/unikraft/catalog-core/blob/main/.gitignore).

razvand reviewed (changes requested):
Update your author and signed-off-by name to be properly written, i.e. use `Dana Caruntu <caruntudana@yahoo.com>` instead of `Dana Caruntu <caruntudana@yahoo.com>`.

razvand reviewed (changes requested):
Add instructions on building the filesystem.

razvand reviewed (changes requested):
Add instructions on building the filesystem.

razvand commented on lua/.gitignore:
```diff
@@ -0,0 +1,2 @@
+/rootfs/
+/workdir/
```
Add ending new line.

razvand commented on lua/Config.uk:
```diff
@@ -0,0 +1,27 @@
+
+	# Select the core Lua runtime library.
+	select LIBLUA
+	# Select the application’s main function entrypoint (Lua interpreter)
+	select LIBLUA_MAIN_FUNCTION
```
```suggestion

	# Select the application’s main function entrypoint (Lua interpreter)
	select LIBLUA_MAIN_FUNCTION

```

Use a blank line before each comment line.

razvand commented on lua/Config.uk:
```diff
@@ -0,0 +1,27 @@
+
+	# Select POSIX environment support for getenv/putenv.
+	select LIBPOSIX_ENVIRON
+	# Use library-parameter parsing for environment variables.
```
```suggestion

	# Use library-parameter parsing for environment variables.
```

razvand reviewed (changes requested):
Update the author (

```text
git config --global user.name 'Dana Caruntu'
git config --global user.email 'caruntudana@yahoo.com'
git commit --amend --reset-author
```

You have two signed-off lines. Leave only one.

Dana-Maria04 committed:
Introduce Lua 5.4 application

Add Lua 5.4 application

Add `lua/` directory:
- `Config.uk`: application configuration
- `Makefile`: for building the application
- `Makefile.uk`: empty placeholder required by the build system
- `README.md`: instructions
- `setup.sh`: script to initialize and populate the root filesystem
- `rootfs/helloworld.lua`: the Lua Hello program
- `fc.x86_64.json` / `fc.arm64.json`: Firecracker configurations for x86_64 and ARM64
- `xen.x86_64.cfg` / `xen.arm64.cfg`: Xen configurations for x86_64 and ARM64
- `.gitignore`: ignore generated files

Signed-off-by: Dana Caruntu <caruntudana@yahoo.com>

razvand reviewed (changes requested):
There is no `rootfs/` directory (with the `helloworld.lua` file inside of it).

razvand commented on lua/xen.arm64.cfg:
```diff
@@ -0,0 +1,5 @@
+name          = "applua"
```
```suggestion
name          = "lua"
```

razvand commented on lua/xen.x86_64.cfg:
```diff
@@ -0,0 +1,5 @@
+name          = "applua"
```
```suggestion
name          = "lua"
```

razvand commented on lua/xen.x86_64.cfg:
```diff
@@ -0,0 +1,5 @@
+name          = "applua"
+vcpus         = "1"
+kernel        = "./workdir/build/lua_xen-x86_64"
```
```suggestion
kernel        = "./workdir/build/lua_xen-x86_64"
ramdisk       = "./initrd.cpio"
cmdline       = "lua_xen-x86_64 vfs.fstab=[ \"initrd0:/:extract::ramfs=1:\" ] -- /helloworld.lua"
```

razvand commented on lua/xen.arm64.cfg:
```diff
@@ -0,0 +1,5 @@
+name          = "applua"
+vcpus         = "1"
+kernel        = "./workdir/build/lua_xen-arm64"
```
```suggestion
kernel        = "./workdir/build/lua_xen-arm64"
ramdisk       = "./initrd.cpio"
cmdline       = "lua_xen-arm64 vfs.fstab=[ \"initrd0:/:extract::ramfs=1:\" ] -- /helloworld.lua"
```

razvand reviewed (changes requested):
Update your commit message to not mention the `.gitignore` update in the message description. Do add a reference to the `.gitignore` file being added. Take inspiration from the commit message introducing `python3-hello`. See [here](https://github.com/unikraft/catalog-core/commit/8a9a5333583f06b0a5a6f0090db1887746ae9768).
